[cli] extend start command to support --connect-only option

librasungirl opened the pull request:
With '--connect-only' option, only establish secure session without attempting to petition to become active commissioner.

lanyuwen reviewed (commented):
Is it possible to amend the usage explanations in the cli README file about this new option?

lanyuwen commented:
I'd also recommend you split the code change of adding state command to a separate PR

librasungirl commented:
added README for the new option, and split `state` cli into separate PR https://github.com/openthread/ot-commissioner/pull/293 which sits on top of this one for dependency. 

@lanyuwen would you please take a review?

wgtdkp commented on src/app/cli/interpreter.cpp:
```diff
@@ -1007,13 +1006,27 @@ Interpreter::Value Interpreter::ProcessStartJob(CommissionerAppPtr &aCommissione
+        if (it != mContext.mCommandKeys.end())
+        {
+            connectOnly = true;
+            error       = aCommissioner->Connect(aExpr[1], port);
```
It seems more straightforward to add a new "connect" command rather than the optional "--connect", given the fact that both the commissioner library and CommissionerApp have the "Connect()" API

librasungirl replied:
It was the initial thought. Changed to this way considering the visually matching start/stop command pairs.

cc @bukepo for vis.

librasungirl commented on src/app/cli/interpreter.cpp:
```diff
@@ -207,7 +207,7 @@ const std::map<std::string, std::string> &Interpreter::mUsageMap = *new std::map
                "config get pskc\n"
                "config set pskc <pskc-hex-string>"},
-    {"start", "start <border-agent-addr> <border-agent-port>\n"
+    {"start", "start <border-agent-addr> <border-agent-port> [--connect]\n"
```
It seems better comparing original `start` command which by default will petition. However it looks a too long string for user to type/input. `--connect` looks to me more straightforwad to start connecting to specified BA on specified port.

librasungirl replied:
no strong preference, updated to `--connect-only`
